libraries/botbuilder-dialogs-adaptive-teams/src/conditions/onTeamsMessagingExtensionSubmitAction.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * @module botbuilder-dialogs-adaptive-teams
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Expression, ExpressionParserInterface } from 'adaptive-expressions';
+import { Channels } from 'botbuilder';
+import { TurnPath } from 'botbuilder-dialogs';
+import { OnInvokeActivity } from 'botbuilder-dialogs-adaptive';
+
+/**
+ * Actions triggered when a Teams InvokeActivity is received with activity.name='composeExtension/submitAction'.
+ */
+export class OnTeamsMessagingExtensionSubmitAction extends OnInvokeActivity {
+    public static $kind = 'Teams.OnMessagingExtensionSubmitAction';
+
+    public getExpression(parser: ExpressionParserInterface): Expression {
+        // if name is 'composeExtension/submitAction'
+        return Expression.andExpression(
+            Expression.parse(
+                `${TurnPath.activity}.channelId == '${Channels.Msteams}' && ${TurnPath.activity}.name == 'composeExtension/submitAction'`
```

Comment:
Quick suggestion would be to add a small helper function somewhere in this package like:

```typescript
function ensureTeamsActivityNamesExpression(names: ...string[]): string {
  return [`${TurnPath.activity}.chanelId = '${Channels.Msteams}`, ...names.map(name => `${TurnPath.activity}.name == '${name}'`)].join(' && ');
}
```

Then you could do:

```typescript
return Expression.andExpression(
  Expression.parse(
    ensureTeamsActivityNamesExpression('compoeExtensions/submitAction')
  )
)
```

Reply:
we spoke about this over Teams, but once implemented, this gets pretty messy due to the # of variations this would need to account for.